impeller/compiler/shader_lib/impeller/texture.glsl

```diff
@@ -10,3 +15,12 @@ vec4 ImpellerTexture(sampler2D texture_sampler,
   }
   return texture(texture_sampler, coords);
 }
+
+// Emulate SamplerAddressMode::ClampToBorder.
+vec4 SampleWithBorder(sampler2D tex, vec2 uv) {
+  float within_bounds = GreaterThan(uv.x, 0) * GreaterThan(uv.y, 0) *
+                        LessThan(uv.x, 1) * LessThan(uv.y, 1);
+  return texture(tex, uv) * within_bounds;
```

Comment:
This looks like it samples transparent black outside of the texture? Which is kDecal, not kClamp.

Reply:
Also, this clamps to 0 right at the edge of the texture, but kDecal will fade to 0 from the center of the edge pixel to the center of the pixel next to it that is outside of the texture.

Reply:
(assuming kLinear interpolation)